core/src/Streamly/Internal/BaseCompat.hs

```diff
@@ -19,7 +19,10 @@ module Streamly.Internal.BaseCompat
 where
 
 import Data.Coerce (Coercible, coerce)
+#if MIN_VERSION_base(4,10,0)
 import Data.Either (fromRight, fromLeft)
+import qualified GHC.Exts as GHCExt
+#endif
```

Comment:
The `#if` condition has to be removed, here and at other places as well. base is always >= 4.10 now.